plugins/cq_cache/cq_cache.py

```diff
@@ -0,0 +1,138 @@
+import cadquery as cq
+import cadquery 
+from cadquery import exporters, importers
+from functools import wraps
+import tempfile
+import os
+import inspect
+
+from OCP.BRepTools import BRepTools
+from OCP.BRep import BRep_Builder
+from OCP.TopoDS import TopoDS_Shape
+
+
+
+
+
+TEMPDIR_PATH = tempfile.gettempdir()
+CACHE_DIR_NAME = "cadquery_geom_cache"
+CACHE_DIR_PATH = os.path.join(TEMPDIR_PATH,CACHE_DIR_NAME)
+CQ_TYPES = [cq.Shape, cq.Solid, cq.Shell, cq.Compound, cq.Face, cq.Wire, cq.Edge, cq.Vertex, TopoDS_Shape, cq.Workplane]
+
+if CACHE_DIR_NAME not in os.listdir(TEMPDIR_PATH):
+    os.mkdir(CACHE_DIR_PATH)
+
+def importBrep(file_path):
+    builder = BRep_Builder()
+    shape = TopoDS_Shape()
+    return_code = BRepTools.Read_s(shape, file_path, builder)
+    if return_code is False:
+        raise ValueError("Import failed, check file name")
+    return shape
+
+
+def get_cache_dir_size(cache_dir_path):
+    total_size = 0
+    for dirpath, dirnames, filenames in os.walk(cache_dir_path):
+        for f in filenames:
+            fp = os.path.join(dirpath, f)
+            total_size += os.path.getsize(fp)
+    return total_size
+
+def delete_oldest_file(cache_dir_path):
+    cwd = os.getcwd()
+    os.chdir(cache_dir_path)
+    files = sorted(os.listdir(os.getcwd()), key=os.path.getmtime)
+    oldest = files[0]
+    os.remove(os.path.join(cache_dir_path,oldest))
+    os.chdir(cwd)
+
+
+def build_file_name(fct, *args, **kwargs):
+    SPACER = "_"
+    file_name = fct.__name__
+    for arg in args:
+        file_name += SPACER + str(arg)
+    for kwarg_value in kwargs.values():
+        file_name += SPACER + str(kwarg_value)
+
+    return file_name
+
+def clear_cq_cache():
+    cache_size = get_cache_dir_size(CACHE_DIR_PATH)
+    for cache_file in os.listdir(CACHE_DIR_PATH):
+        os.remove(os.path.join(CACHE_DIR_PATH, cache_file))
+    print(f"Cache cleared for {round(cache_size*1e-6,3)} MB ")
+
+def using_same_function(fct, file_name):
+    with open(file_name,"r") as f :
+        cached_function = "".join(f.readlines()[:-1])
+
+    caching_function = inspect.getsource(fct)
+    if cached_function == caching_function:
+        return True 
+    else :
+        return False
```

Comment:
This code on stackexchange might be of interest to you: https://codereview.stackexchange.com/questions/192848/python-function-to-hash-a-python-function

The author does a similar task but he ignores comments, docstrings and minor formatting changes. Also stores the result as a small hash instead of the complete code.

It's a fair bit of extra work though, it's up to you if you want to implement it.

Reply:
It doesn't seems too complicated so I might give it a try. Even though I'm not sure it will make more time gain overall than the one I'm going to take adding this 😄

Reply:
Yeah, but I'll give you style points for implementing it. :rofl: